src/main/java/com/suse/saltstack/netapi/parser/SaltStackParser.java

```diff
@@ -50,6 +60,25 @@ public T parse(InputStream inputStream) throws SaltStackParsingException {
         Reader streamReader = new BufferedReader(inputStreamReader);
 
         // Parse result type from the returned JSON
-        return new Gson().fromJson(streamReader, type.getType());
+        return gson.fromJson(streamReader, type.getType());
+    }
+
+    /**
+     * Deserializer for date representation received from the API
+     * (which represents it as a (floating) number of seconds since the Epoch).
+     */
+    private class SaltStackDateDeserializer implements JsonDeserializer<Date> {
+        @Override
+        public Date deserialize(JsonElement jsonElement, Type type,
+                JsonDeserializationContext jsonDeserializationContext)
+                throws JsonParseException {
+            try {
+                double dateMiliSecs = jsonElement.getAsDouble() * 1000;
+                return new Date((long) dateMiliSecs);
+            } catch (Exception e) {
+                e.printStackTrace(); // TODO ideally use some logging infrastructure
```

Comment:
Please throw an appropriate `JsonParseException` here in case the format is not as expected and do not swallow other unexpected exceptions.